### PR TITLE
Nuxt

### PR DESCRIPTION
**Pull request by dfcook**
Move the entire site to nuxt.js to enable SSR

### CODE REVIEW

**Comment by jasonify** [comment]
Notes:
Seems to require node 8.10

Also seeing an error for posts with comments. 
`
        return this.me._id === this.comment.author._id
`

File: package.json [hunk 1]
```diff
@@ -1,131 +1,59 @@
...
+    "dev:api-test": "cross-env API_ENV=staging npm run nuxt",
+    "build": "nuxt build",
+    "build:staging": "cross-env API_ENV=staging npm run build",
+    "start": "nuxt start",
+    "generate": "nuxt generate",
+    "lint": "eslint --ext .js,.vue --ignore-path .gitignore .",
+    "lint:fix": "eslint --ext .js,.vue --ignore-path .gitignore . --fix",
+    "test": "ava --serial --verbose",
+    "precommit": "npm run lint"
+  },
```
Comment (jasonify) [inline package.json]: We might want:   "heroku-postbuild": "npm run build"
per the heroku nuxt guide: https://nuxtjs.org/faq/heroku-deployment

**Comment by jasonify** [comment]
We probably want to remove yarn.lock too, since heroku doesn't seem to like having that and package.json

**Comment by dfcook** [comment]
Added these changes

**Comment by jasonify** [comment]
This is looking great :)

Just two observations as I was testing:

1) Mobile post/ view seems to have a vertical scroll bar for me. Maybe we can remove the left most side details and it would help?

2) I saw what looked like a red errors notification floating that then disappeared before I could read it. Not sure if we could either a) let the user close it 2) leave it there a bit longer 3) console.log the error too so it can be used for debugging later?

**Comment by jasonify** [comment]
Hey, just another quick note:

Seeing a small issue with the config file getting used, seems to be partially be working. 

When I run `npn run dev` I'm seeing that it hits the correct endpoint  (seems to be set inside `nuxt.config.js`) but when we are pulling the config settings in something like our actions, we are getting `undefined` passed as the `env` which defaults to `prod`. So when I tested out the Subscription part of the app it is trying to use the prod key for stripe (but the rest of the app is pointing to dev). Not entirely terrible since in prod it would in theory all work.



@joshmorel @andrewmarklloyd  sounds like you guys might know this part the best. Let me know, I can take a look too.

**Comment by jasonify** [comment]
For now we're trying an alternative to nuxt so closing this for now.

**Pull request closed without merging**